jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/EcsExplorerNodes.kt

```diff
@@ -53,16 +62,38 @@ class EcsClusterNode(project: Project, private val clusterArn: String) :
         .getResourceNow(EcsResources.listServiceArns(clusterArn))
         .map { nodeProject.getResourceNow(EcsResources.describeService(clusterArn, it)) }
         .map { EcsServiceNode(nodeProject, it, clusterArn) }
+
+    override val cloudApiResourceType = "AWS::ECS::Cluster"
+    override fun identifier() = clusterArn
 }
 
-class EcsServiceNode(project: Project, private val service: Service, private val clusterArn: String) :
+class EcsServiceNode(project: Project, internal val service: Service, private val clusterArn: String) :
     AwsExplorerResourceNode<Service>(project, EcsClient.SERVICE_NAME, service, AwsIcons.Resources.Ecs.ECS_SERVICE),
-    ResourceLocationNode {
+    ResourceLocationNode,
+    CloudApiResource {
 
     override fun resourceType() = "service"
     override fun resourceArn(): String = value.serviceArn()
     override fun displayName(): String = value.serviceName()
     override fun location() = EcsCloudDebugLocation(nodeProject, service)
-    fun executeCommandEnabled() = value.enableExecuteCommand()
+    fun executeCommandEnabled(): Boolean = value.enableExecuteCommand()
     fun clusterArn(): String = clusterArn
+
+    override val cloudApiResourceType = "AWS::ECS::Service"
+    override fun identifier() = "${resourceArn()}|$clusterArn"
+}
+
+class ViewTaskDefinitionAction(private val node: EcsServiceNode) : AnAction(message("dynamic_resources.view_configuration_action_title", "Task Definition")) {
+    override fun actionPerformed(e: AnActionEvent) {
+        val identifier = DynamicResourceIdentifier(node.nodeProject.getConnectionSettingsOrThrow(), "AWS::ECS::TaskDefinition", node.service.taskDefinition())
+        DynamicResourceFileManager.getInstance(node.nodeProject).openEditor(identifier)
+    }
+}
+
+class EcsTaskDefinitionActionContributor : AwsExplorerActionContributor {
```

Comment:
Could this action be added to the action group directly through `plugin.xml`?

Reply:
Could be - but I wanted `View Service Configuration` and `View Task Definition Configuration` to appear near each other (next to each other).